base/logging.c

```diff
@@ -82,15 +82,18 @@ gchar *
 get_time (gchar *time_fmt)
 {
   time_t now;
-  struct tm *ts;
+  struct tm ts;
   gchar buf[80];
 
   /* Get the current time. */
   now = time (NULL);
 
   /* Format and print the time, "ddd yyyy-mm-dd hh:mm:ss zzz." */
-  ts = localtime (&now);
-  strftime (buf, sizeof (buf), time_fmt, ts);
+  if (localtime_r (&now, &ts) == NULL)
+    {
+      g_warning ("%s: localtime failed", __func__);
```

Comment:
I'm not sure if we should log a warning here because `get_time` is used in the log function, so this may result in an infinite recursion if the problem with the `localtime_r` call persists.